timm/models/hub.py

```diff
@@ -130,25 +130,28 @@ def push_to_hf_hub(
     revision=None,
     model_config=None,
 ):
+    if isinstance(use_auth_token, str):
+        token = use_auth_token
+    else:
+        token = HfFolder.get_token()
+    if token is None:
+        raise ValueError(
+            "You must login to the Hugging Face hub on this computer by typing `huggingface-cli login` and "
+            "entering your credentials to use `use_auth_token=True`. Alternatively, you can pass your own "
+            "token as the `use_auth_token` argument."
+        )
+
     if repo_namespace_or_url:
         repo_owner, repo_name = repo_namespace_or_url.rstrip('/').split('/')[-2:]
     else:
-        if isinstance(use_auth_token, str):
-            token = use_auth_token
-        else:
-            token = HfFolder.get_token()
-
-        if token is None:
-            raise ValueError(
-                "You must login to the Hugging Face hub on this computer by typing `transformers-cli login` and "
-                "entering your credentials to use `use_auth_token=True`. Alternatively, you can pass your own "
-                "token as the `use_auth_token` argument."
-            )
-
         repo_owner = HfApi().whoami(token)['name']
         repo_name = Path(local_dir).name
 
-    repo_url = f'https://huggingface.co/{repo_owner}/{repo_name}'
+    repo_id = f'{repo_owner}/{repo_name}'
+    repo_url = f'https://huggingface.co/{repo_id}'
+
+    # Create repo if doesn't exist yet
+    HfApi().create_repo(repo_id, token=use_auth_token, exist_ok=True)
```

Comment:
Should we add `private` kwarg while we are at it? (not necessary, just a thought).

Reply:
I can easily add it as well indeed.

In general, I thought that we could really simplify the `hfh` integration by pushing the model without using the `Repository` class. It would ease things by requiring only a model, a repo_id, a revision and a token.

So changing the signature from:
```py
def push_to_hf_hub(
    model,
    local_dir,
    repo_namespace_or_url=None,
    commit_message='Add model',
    use_auth_token=True,
    git_email=None,
    git_user=None,
    revision=None,
    model_config=None,
):
```

To:
```py
def push_to_hf_hub(
    model,
    repo_id: str,
    commit_message: str ='Add model',
    token: Optional[str] = None,
    revision: Optional[str] =None,
    model_config: Optional[dict] = None,
):
``` 

And inside the function create a temporary folder to dump the model and then use the http-based `upload_folder`. A few benefits I see:
- less verbose signature
- **No need to clone the repo** before pushing (huge improvement for bandwidth)
- No local folder created on machine
- would be easy to add `private` or `create_pr` attribute

Of course, it comes with its drawback, especially breaking changes in the function signature. I am really not familiar with timm so please consider this as a suggestion. I can help on the implementation but will let you discuss if that is something you want :)

Reply:
and yea, this integration was written a long time ago (a year ago at least I think?) so its quite out of date. newer fn's would be so much easier on the eyes.

Reply:
If you'd like to enhance/clean it up, would be good to clean it up in this PR...

Reply:
@Wauplin do you want to, or would you like me to work on top of what you've done here?

Reply:
@rwightman @nateraw I just refactored the push_to_hub helper in https://github.com/rwightman/pytorch-image-models/pull/1547/commits/9b114754dbbf2fc327bd7835af91aa41af7348d3.

I tested it here: https://huggingface.co/Wauplin/resnet18-random-classifier-bug-0-11-0/tree/main
Please let me know if that would be ok for you. Also is there some tests to adapt ? (I haven't found one)